src/Microsoft.Azure.WebJobs.Host/Blobs/Triggers/BlobTriggerBinding.cs

```diff
@@ -252,10 +235,10 @@ private static IAsyncObjectToTypeConverter<IStorageBlob> CreateConverter(IStorag
                 throw new InvalidOperationException("Unable to convert trigger to IStorageBlob.");
             }
 
-            Host.Bindings.IValueProvider valueProvider = await _argumentBinding.BindAsync(conversionResult.Result, context);
+            // Leave ValueProvider null - other binding rules will take care of it.
```

Comment:
this comment seems like it should just be removed, since it applies to the code a few lines below. Perhaps you should add a new ctor to TriggerData that doesn't take a value provider, and put comments there describing that scenario?